src/foam/nanos/ruler/RuleActionMergeDecorator.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'RuleActionMergeDecorator',
+
+  documentation: `
+    Merge calls occurring within mergeDelay period to the delegate rule action.
+
+    Usage: Add RuleActionMergeDecorator to the 'action' of an async rule to
+    merge repeated calls to the underlying rule action. For example,
+
+      {
+        class: 'foam.nanos.ruler.Rule',
+        ...,
+        async: true,
+        action: {
+          class: 'foam.nanos.ruler.RuleActionMergeDecorator',
+          mergeDelay: 1000,
+          delegate: {
+            class: 'MyRuleAction'
+          }
+        }
+      }
+  `,
+
+  implements: [ 'foam.nanos.ruler.RuleAction' ],
+
+  javaImports: [
+    'foam.nanos.logger.Logger',
+    'java.util.Timer',
+    'java.util.TimerTask',
+    'java.util.concurrent.ConcurrentHashMap'
+  ],
+
+  properties: [
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.ruler.RuleAction',
+      name: 'delegate'
+    },
+    {
+      class: 'Int',
+      name:  'mergeDelay',
+      units: 'ms'
+    },
+    {
+      class: 'Object',
+      javaType: 'java.util.Timer',
+      name: 'timer',
+      javaFactory: 'return new Timer();'
```

Comment:
I don't like this because it is going to create one thread per merged rule action.
I would rather add a schedule() method to our Agency interface which is the same as the current submit() method but takes a delay. It would use a single internal Timer for the whole system to wait for the specified delay before then calling submit() on the task to be executed.

Reply:
Fixed in https://github.com/kgrgreer/foam3/pull/1027/commits/997de326e43b3dac06892cd65d6c9de8b1a915dc.